### PR TITLE
[gettext-libintl] Link CoreFoundation on apple

### PR DESCRIPTION
**Pull request by dg0yt**
Cherry-picked from https://github.com/microsoft/vcpkg/pull/38658.

### CODE REVIEW

**Comment by dg0yt** [comment]
Side note:
The single list item `"-framework CoreFoundation"` avoids absolute SDK paths while building in CMake, and is used elsewhere in vcpkg. But the embedded space as linker option separator isn't entirely accurate and may cause problems when link libs are interpreted by other programs. 

TIL https://cmake.org/cmake/help/latest/command/find_library.html:
 > If the library found is a framework, then `<VAR>` will be set to the full path to the framework `<fullPath>/A.framework`. When a full path to a framework is used as a library, CMake will use a `-framework A`, and a `-F<fullPath>` to link the framework to the target.

Until the CMake behavior is better covered in vcpkg's check and fixup of absolute paths, I will stick to the established practice.

File: ports/gettext-libintl/vcpkg-cmake-wrapper.cmake [hunk 1]
```diff
@@ -14,4 +14,10 @@ if(Intl_FOUND AND Intl_LIBRARIES)
             set_property(TARGET Intl::Intl APPEND PROPERTY INTERFACE_LINK_LIBRARIES $<LINK_ONLY:Iconv::Iconv>)
         endif()
     endif()
+    if(APPLE)
+        list(APPEND Intl_LIBRARIES "-framework CoreFoundation")
```
Comment (Neumann-A) [inline ports/gettext-libintl/vcpkg-cmake-wrapper.cmake:18]: Can something like `-Wl,-framework,CoreFoundation`  work?
Don't know if `"SHELL: -framework CoreFoundation"` is also an option or if it only works for `add_compile_options`

Comment (Neumann-A) [reply ports/gettext-libintl/vcpkg-cmake-wrapper.cmake:18]: I was more concerned about something in cmake transforming `"-framework CoreFoundation"` to `"-framework; CoreFoundation"` and then deciding to reorder the flags. However, if there are no regressions in CI it probably works without any change considering how deep down intl typically is.

**Review by WangWeiLin-MV — APPROVED** [review]
The port usage tests pass with the following triplets:
* x64-osx
* x64-windows

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**